File: IQ/Classes/clsProductType.vb
Imports dataAccess

Public Class clsProductType

    Property ID As Integer
    Property Code As String
    Property Translation As clsTranslation
    Property Order As Short

    Dim oCode As String

    Public Sub New(ByVal code As String, ByVal translation As clsTranslation, order As Short)

        Dim sql$
        sql$ = "INSERT INTO ProductType (code,fk_Translation_key_text,[order]) VALUES ('" & code & "'," & translation.Key & "," & order & ");"
        Me.ID = da.DBExecutesql(sql$, True)
        Me.Code = code
        Me.Translation = translation
        Me.Order = order

        iq.ProductTypes.Add(Me.ID, Me)
        iq.i_ProductType_Code.Add(Me.Code, Me)

        oCode = Me.Code

    End Sub
    Public ReadOnly Property DisplayName(Language As clsLanguage) As String
        Get
            DisplayName = Me.Code & " " & Me.Translation.text(Language)
        End Get
    End Property

    Public Sub New()

    End Sub

    Public Function Insert()

        Return New clsProductType(Me.Code, Me.Translation, 0)

    End Function

    Public Sub Update()

        Dim sql$
        sql$ = "UPDATE [ProductType] SET code=" & da.SqlEncode(Me.Code) & ",fk_translation_key_text=" & Me.Translation.Key & ",[order]=" & Order & " WHERE ID=" & Me.ID

        Try
            iq.i_ProductType_Code.Remove(oCode)
            iq.i_ProductType_Code.Add(Me.Code, Me)
            da.dbexecutesql(sql$)

        Catch ex As System.Exception
            Stop ' probably a duplictae code
        End Try

    End Sub

    Public Sub Delete()

        Dim sql$
        sql$ = "DELETE FROM [ProductType] WHERE ID=" & Me.ID
        da.dbexecutesql(sql$)

    End Sub

    Public Sub New(ByVal ID As Integer, ByVal code As String, ByVal translation As clsTranslation, order As Short)

        Me.ID = ID
        Me.Code = code
        Me.Translation = translation
        Me.Order = order

        iq.ProductTypes.Add(Me.ID, Me)
        iq.i_ProductType_Code.Add(Me.Code, Me)

        oCode = Me.Code

    End Sub


End Class

File: IQ/Classes/clsProductType.cs
using dataAccess;


public class clsProductType
{

    public int ID { get; set; }
    public string Code { get; set; }
    public clsTranslation Translation { get; set; }
    public short Order { get; set; }

    string oCode;

    public clsProductType(string code, clsTranslation translation, short order)
    {

        object sql = null;
        sql = "INSERT INTO ProductType (code,fk_Translation_key_text,[order]) VALUES (\'" + code + "\'," + Translation.Key + "," + System.Convert.ToString(order) + ");";
        this.ID = System.Convert.ToInt32(da.DBExecutesql(sql, true));
        this.Code = code;
        this.Translation = translation;
        this.Order = order;

        iq.ProductTypes.Add(this.ID, this);
        iq.i_ProductType_Code.Add(this.Code, this);

        oCode = this.Code;

    }
    public string get_DisplayName(clsLanguage Language)
    {
        string returnValue = "";
        returnValue = this.Code + " " + this.Translation.text(Language);
        return returnValue;
    }

    public clsProductType()
    {

    }

    public dynamic Insert()
    {

        return new clsProductType(this.Code, this.Translation, (short)0);

    }

    public void Update()
    {

        object sql = null;
        sql = "UPDATE [ProductType] SET code=" + da.SqlEncode(this.Code) + ",fk_translation_key_text=" + this.Translation.Key + ",[order]=" + System.Convert.ToString(Order) + " WHERE ID=" + System.Convert.ToString(this.ID);

        try
        {
            iq.i_ProductType_Code.Remove(oCode);
            iq.i_ProductType_Code.Add(this.Code, this);
            da.dbexecutesql(sql);

        }
        catch (System.Exception)
        {
            Debugger.Break(); // probably a duplictae code
        }

    }

    public void Delete()
    {

        object sql = null;
        sql = "DELETE FROM [ProductType] WHERE ID=" + System.Convert.ToString(this.ID);
        da.dbexecutesql(sql);

    }

    public clsProductType(int ID, string code, clsTranslation translation, short order)
    {

        this.ID = ID;
        this.Code = code;
        this.Translation = translation;
        this.Order = order;

        iq.ProductTypes.Add(this.ID, this);
        iq.i_ProductType_Code.Add(this.Code, this);

        oCode = this.Code;

    }


}